src/components/ui/datefield.tsx

```diff
@@ -0,0 +1,72 @@
+import { useTranslation } from "react-i18next";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface DateFieldProps {
+  value?: string;
+  onChange?: (value: string) => void;
+}
+
+export default function DateField({ value = "", onChange }: DateFieldProps) {
+  const [year, month, day] = value.split("-");
+  const { t } = useTranslation();
+
+  const handleDayChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newDay = e.target.value;
+    if (onChange) {
+      onChange(`${year || ""}-${month || ""}-${newDay}`);
+    }
+  };
```

Comment:
the implementation in existing PatientRegistration form had auto focus change once a valid value is inputted, can we preserve that logic in this too?

ref: https://github.com/ohcnetwork/care_fe/blob/23806f6088ca13a9c20498efcc4de60af255a215/src/components/Patient/PatientRegistration.tsx#L570-L580